src/com/mambu/apisdk/services/AccountingService.java

```diff
@@ -234,23 +234,97 @@ public GLAccount getGLAccount(String glCode, String fromDate, String toDate) thr
 		// /api/gljournalentries
 		// See MBU-1737
 
-		if (entries == null || entries.size() < 2) {
-			throw new IllegalArgumentException("At least one debit and one credit entry is required");
-		}
-		if (date == null) {
-			throw new IllegalArgumentException("Date must not be null");
+		validateParamsForPostingGLEntries(entries, date);
+
+		ParamsMap params = populateBaseParams(entries, branchId, date, notes);
+
+		return executePostGLJournalEntries(params);
+	}
+
+	/**
+	 * Post GL Journal Entries
+	 * 
+	 * @param entries
+	 *            a list of entries with the GL transaction details. Must not be null. At least one debit and one credit
+	 *            entry must be specified. Any number of journal entries may be posted with a given date and branch id
+	 *            as long as the standard accounting rules apply. For each entry its glCode, entryType and amount must
+	 *            not be null.
+	 * @param branchId
+	 *            a branch id.
+	 * @param date
+	 *            The date of the posting of the journal entry. Must be not null
+	 * @param notes
+	 *            transaction notes
+	 * @param transactionId
+	 *            the transaction id
+	 * @return created journal entries
+	 * @throws MambuApiException
+	 */
+	public List<GLJournalEntry> postGLJournalEntries(List<ApiGLJournalEntry> entries, String branchId, String date,
+			String notes, String transactionId) throws MambuApiException {
+
+		// POST
+		// "branchId=2&date=2010-02-03&debitAccount1=100001&debitAmount1=30&creditAccount1=100002&creditAmount1=30&transactionID=9284"
+		// /api/gljournalentries
+		// See MBU-15973
+
+		validateParamsForPostingGLEntries(entries, date);
+
+		ParamsMap params = populateBaseParams(entries, branchId, date, notes);
+
+		if (transactionId != null) {
+			params.put(APIData.TRANSACTION_ID, transactionId);
 		}
 
+		return executePostGLJournalEntries(params);
+	}
+
+	/**
+	 * Executes the posting of GL journal entries
+	 * 
+	 * @param params
+	 *            the parameters of the call
+	 * @return a list containing successful posted entries
+	 * 
+	 * @throws MambuApiException
+	 */
+	private List<GLJournalEntry> executePostGLJournalEntries(ParamsMap params) throws MambuApiException {
+
+		// Create ApiDefinition
+		ApiDefinition apiDefiinition = new ApiDefinition(APIData.GLJOURNALENTRIES, ContentType.WWW_FORM, Method.POST,

```

Comment:
apiDefiinition - I know that this was converted to a stand-alone method, but please correct the variable name